torchqdynamics/utils/utils.py

```diff
@@ -22,95 +22,106 @@
 ]
 
 
-def is_ket(x: Tensor) -> Tensor:
-    """Check if a tensor is in state vector format."""
+def is_ket(x: Tensor) -> bool:
+    """Check if a tensor is in state vector format.
+
+    Args:
+        x ( Tensor (..., d, n) ): State vector, linear map or density matrix.
+
+    Returns:
+        `True` if the last dimension `n` is `1`, `False` otherwise.
+    """
     return x.size(-1) == 1
 
 
 def ket_to_bra(x: Tensor) -> Tensor:
-    """Linear map (bra) representation of a state vector (ket).
+    r"""Linear map $\bra\psi$ representation of a state vector $\ket\psi$.
 
     Args:
-        x: Tensor of dimension `(..., d, 1)`.
+        x ( Tensor (..., d, 1) ): State vector.
 
     Returns:
-        Tensor of dimension `(..., 1, d)`.
+        ( Tensor (..., 1, d) ): Corresponding linear map.
```

Comment:
Same comment, I don't think this deserves a Returns section. See e.g. https://pytorch.org/docs/stable/generated/torch.adjoint.html?highlight=adjoint#torch.adjoint. Just having the typing at the top in the function definition `torch.adjoint(Tensor) → Tensor` is enough IMO. Otherwise it's too heavy and harder to understand.

Arguably, even the Args section is not useful here if an Example section is provided, but that's a bit more extreme.

Reply:
Yes I agree, it's just a little annoying I don't know where to place the tensors shape 🤔